Model/review.model.js

```diff
@@ -1,5 +1,38 @@
 const db = require("../db/connection.js");
 
+exports.selectReviews = (query) => {
+  const columnName = Object.keys(query)[0];
+  const whereValue = query[columnName];
+
+  let queryStr =
+    "SELECT reviews.*, COUNT(comment_id) AS comment_count FROM reviews LEFT JOIN comments ON comments.review_id = reviews.review_id";
+  if (columnName) {
+    queryStr += ` WHERE ${columnName} = $1`;
+  }
+
+  queryStr += " GROUP BY reviews.review_id;";
+
+  if (whereValue) {
+    return db.query(queryStr, [whereValue]).then((results) => {
+      if (results.rows.length === 0) {
+        return db
+          .query(`SELECT * FROM reviews WHERE ${columnName} = $1`, [whereValue])
+          .then((columnExists) => {
+            if (!columnExists.rows.length > 0) {
+              return Promise.reject({ status: 404, msg: "Not Found" });
+            }
+            return results.rows;
+          });
+      }
+      return results.rows;
+    });
```

Comment:
This is nesting promises. We try not to do this because it makes it more difficult to read. Return the promise, chain the .then on the outermost promise. Since we need to check two things at the same time, a better way to do this would be to store the comments query and the review query in an array and use a Promise.all(), this way we can just check the rows on the review result and if its 0, reject it, else just return the comments